lstests/core/service.py

```diff
@@ -0,0 +1,1526 @@
+        return "defproducer" + string.ascii_lowercase[num] if num < 26 else "defpr" + int_to_base26(8031810176 + num)[1:]
+
+# --------------- check and verify ------------------------------------------------------------------------------------
+
+    def verify(self,
+               transaction_id,
+               verify_key=None,
+               verify_async=False,
+               dont_raise=False,
+               retry=None,
+               sleep=None,
+               level=None,
+               retry_level=None,
+               error_level=None,
+               buffer=None):
+        verify_async = helper.override(self.verify_async, verify_async, self.cla.verify_async)
+        retry = helper.override(self.verify_retry, retry, self.cla.verify_retry)
+        sleep = helper.override(self.verify_sleep, sleep, self.cla.verify_sleep)
+        verified = False
+        for __ in range(retry + 1):
+            if self.verify_transaction(transaction_id=transaction_id, verify_key=verify_key, level=retry_level, buffer=buffer):
+                verified = True
+                break
+            time.sleep(sleep)
+        if verify_async:
+            self.print_header("async verify transaction", level=level, buffer=True)
+            if verified:
+                self.log(color.black_on_green(f"{verify_key.title()}") + f" {transaction_id}", level=level, buffer=True)
+            else:
+                self.log(color.black_on_red(f"Not {verify_key.title()}") + f" {transaction_id}", level=error_level, buffer=True)
+            self.print_header("", sep="", level=level, buffer=True)
+            self.flush()
+        else:
+            if verified:
+                self.log(color.black_on_green(f"{verify_key.title()}"), level=level)
+            else:
+                self.log(color.black_on_red(f"Not {verify_key.title()}"), level=error_level)
+        if not verified and not dont_raise:
+            raise BlockchainError(f"{transaction_id} cannot be verified")
+        return verified
+
+    def check_sync(self, retry=None, sleep=None, min_sync_count=None, max_block_lag=None, dont_raise=False, level="debug"):
+        @dataclasses.dataclass
+        class SyncResult:
+            in_sync: bool
+            sync_count: int
+            min_block_num: int
+            max_block_num: int = None
+            def __post_init__(self):
+                if self.in_sync:
+                    self.block_num = self.max_block_num = self.min_block_num
+                else:
+                    assert self.min_block_num == math.inf or self.min_block_num <= self.max_block_num
+                    self.block_num = -1
+        retry = helper.override(self.sync_retry, retry, self.cla.sync_retry)
+        sleep = helper.override(self.sync_sleep, sleep, self.cla.sync_sleep)
+        min_sync_count = helper.override(self.node_count, min_sync_count)
+        self.print_header("check sync", level=level)
+        for __ in range(retry + 1):
```

Comment:
Another double underscore.

Reply:
Now back to the single underscore. Thank you.